core-tests/src/test/java/org/mule/runtime/core/internal/profiling/consumer/tracing/DefaultCoreEventTracerTestCase.java

```diff
@@ -129,6 +132,39 @@ public void endCurrentExecutionSpan() {
     verify(distributedTraceContext).endCurrentContextSpan();
   }
 
+  @Test
```

Comment:
Can we parametrize the tests you added so we can check that the system property that propagates the errors also works?